### PR TITLE
[#638] Use Formatting options passed by the client

### PR DESCRIPTION
**Pull request by cdietrich**
[#638] Use Formatting options passed by the client
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
redo of https://github.com/eclipse/xtext-core/pull/642

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/formatting/FormattingService.xtend [hunk 1]
```diff
@@ -60,12 +68,32 @@ class FormattingService {
...
+	def List<TextEdit> format(XtextResource resource, Document document, int offset, int length, FormattingOptions options) {
+		var indent = "\t"
+		if (options !== null) {
+			if (options.insertSpaces) {
+				indent = Strings.padEnd("", options.tabSize," ")
+			}
+		}
 		if (formatter2Provider !== null) {
-			val replacements = format2(resource, new TextRegion(offset, length), null)
+			val replacements = format2(resource, new TextRegion(offset, length), new MapBasedPreferenceValues(#{"indentation"->indent}))
```
Comment (JanKoehnlein) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/formatting/FormattingService.xtend]: Why do you create an immutable map here? 
Makes my code fail when trying to add additional values.

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/formatting/FormattingService.xtend]: where do you try to add additional stuff?

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/formatting/FormattingService.xtend]: besides this here is no intention behind this. so feel free to change

**Pull request merged**